algviz/graphs/graphs.py

```diff
@@ -0,0 +1,36 @@
+import pygraphviz as pgv
+
+
+def nodesToLocations(nodes, edges, algo="dot"):
+    """ Maps node to their coordinates.
+    """
+    G = pgv.AGraph()
+    G.graph_attr['fixedsize'] = True
+    for node in nodes:
+        G.add_node(node, fixedsize=True, width=node.width, height=node.height)
+    for (node1, node2) in edges:
+        G.add_edge(node1, node2)
+    G.layout(prog=algo)
+    dictNodesLocations = {}
+    for node in G.nodes():
+        dictNodesLocations[node.get_name()] = node.attr['pos']
+    return dictNodesLocations
+
+
+class Node:
```

Comment:
I'm not convinced that the node class is necessary. I think it would be better design to just pass name, width, height to the nodeToLocations function directly as arguments.